hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -320,11 +320,9 @@ public void refreshTimeline() throws IOException {
             .build();
         switch (meta.getTableType()) {
           case COPY_ON_WRITE:
-            this.commitTimelineOpt = Option.of(meta.getActiveTimeline().getCommitTimeline().filterCompletedInstants());
-            this.allCommitsTimelineOpt = Option.of(meta.getActiveTimeline().getAllCommitsTimeline());
-            break;
           case MERGE_ON_READ:
-            this.commitTimelineOpt = Option.of(meta.getActiveTimeline().getDeltaCommitTimeline().filterCompletedInstants());
+            // we can use getCommitsTimeline for both COW and MOR here, because for COW there is no deltacommit
+            this.commitsTimelineOpt = Option.of(meta.getActiveTimeline().getCommitsTimeline().filterCompletedInstants());
```

Comment:
This makes MOR to consider additional `COMMIT_ACTION` (compaction) and `REPLACE_COMMIT_ACTION`.  Is this correct?

Reply:
Yes, that's correct. 
In the case of changing COW to MOR, the checkpoint saved in the `.commit` file wrote by COW will be lost when continue with MOR.

Reply:
In that case, for MOR table, is the data from compaction commit also pulled in during incremental pulls, which can cause duplication?

Reply:
As part of this change, we will always use deltacommit only timeline when one is found in the destination timeline (non-empty).